addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -75,6 +75,12 @@ odoo.define('website_form.s_website_form', function (require) {
             // Empty imputs
             this.$target.find('input[type="text"], input[type="email"], input[type="number"], textarea').val('');
 
+            // Apply default values
+            this.$target[0].querySelectorAll('input[type="text"], input[type="email"], input[type="number"]').forEach(el => {
+                el.value = el.getAttribute('value');
+            });
+            this.$target[0].querySelectorAll('textarea').forEach(el => el.value = el.textContent);
```

Comment:
Is not your first commit meant to avoid having to do that?

Reply:
This is the website_form widget not the options. This is unavoidable as I remove all value (property) from the form's inputs before edition to remove user's inputs.